checkSlowWriterTxAbortsOnHoleFill unit test is failing

## Overview

checkSlowWriterTxAbortsOnHoleFill unit test is failing constantly on local runs

Looks like some stream log issue.

```
Last 25 Logs Until Failure
19:00:02.061 ERROR [LogUnit-Write-Processor-0] o.c.i.BatchWriter - Stream log error. Batch [queue size=0]. StreamLog: [trim mark: 0, tails: TailsResponse(logTail=0, streamTails={})].
 org.corfudb.runtime.exceptions.OverwriteException: null
	at org.corfudb.infrastructure.log.InMemoryStreamLog.throwLogUnitExceptionsIfNecessary(InMemoryStreamLog.java:97)
	at org.corfudb.infrastructure.log.InMemoryStreamLog.append(InMemoryStreamLog.java:60)
	at org.corfudb.infrastructure.BatchWriter.batchWriteProcessor(BatchWriter.java:250)
19:00:02.062 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - Recover[Token(epoch=0, sequence=0)]: read chain head 1/1
 19:00:02.063 WARN  [test-main] o.c.r.v.StreamsView - append[0]: Overwritten after 0 retries, streams [bee9]
 19:00:12.145 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - fillHole[Token(epoch=0, sequence=1)]: chain head 1/1
 19:00:12.146 INFO  [test-main] o.c.r.v.s.BackpointerStreamView - followBackpointers[bee9@-1]: Found hole at address 1. Starting single step downgrade.
 19:00:12.148 ERROR [LogUnit-Write-Processor-0] o.c.i.BatchWriter - Stream log error. Batch [queue size=0]. StreamLog: [trim mark: 0, tails: TailsResponse(logTail=1, streamTails={})].
 org.corfudb.runtime.exceptions.OverwriteException: null
	at org.corfudb.infrastructure.log.InMemoryStreamLog.throwLogUnitExceptionsIfNecessary(InMemoryStreamLog.java:97)
	at org.corfudb.infrastructure.log.InMemoryStreamLog.append(InMemoryStreamLog.java:60)
	at org.corfudb.infrastructure.BatchWriter.batchWriteProcessor(BatchWriter.java:250)
19:00:12.149 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - Recover[Token(epoch=0, sequence=1)]: read chain head 1/1
 19:00:12.150 WARN  [test-main] o.c.r.v.StreamsView - append[1]: Overwritten after 1 retries, streams [bee9]
 19:00:22.232 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - fillHole[Token(epoch=0, sequence=2)]: chain head 1/1
 19:00:22.233 INFO  [test-main] o.c.r.v.s.BackpointerStreamView - followBackpointers[bee9@-1]: Found hole at address 2. Starting single step downgrade.
 19:00:22.234 ERROR [LogUnit-Write-Processor-0] o.c.i.BatchWriter - Stream log error. Batch [queue size=0]. StreamLog: [trim mark: 0, tails: TailsResponse(logTail=2, streamTails={})].
 org.corfudb.runtime.exceptions.OverwriteException: null
	at org.corfudb.infrastructure.log.InMemoryStreamLog.throwLogUnitExceptionsIfNecessary(InMemoryStreamLog.java:97)
	at org.corfudb.infrastructure.log.InMemoryStreamLog.append(InMemoryStreamLog.java:60)
	at org.corfudb.infrastructure.BatchWriter.batchWriteProcessor(BatchWriter.java:250)
19:00:22.235 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - Recover[Token(epoch=0, sequence=2)]: read chain head 1/1
 19:00:22.235 WARN  [test-main] o.c.r.v.StreamsView - append[2]: Overwritten after 2 retries, streams [bee9]
 19:00:32.306 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - fillHole[Token(epoch=0, sequence=3)]: chain head 1/1
 19:00:32.308 INFO  [test-main] o.c.r.v.s.BackpointerStreamView - followBackpointers[bee9@-1]: Found hole at address 3. Starting single step downgrade.
 19:00:32.310 ERROR [LogUnit-Write-Processor-0] o.c.i.BatchWriter - Stream log error. Batch [queue size=0]. StreamLog: [trim mark: 0, tails: TailsResponse(logTail=3, streamTails={})].
 org.corfudb.runtime.exceptions.OverwriteException: null
	at org.corfudb.infrastructure.log.InMemoryStreamLog.throwLogUnitExceptionsIfNecessary(InMemoryStreamLog.java:97)
	at org.corfudb.infrastructure.log.InMemoryStreamLog.append(InMemoryStreamLog.java:60)
	at org.corfudb.infrastructure.BatchWriter.batchWriteProcessor(BatchWriter.java:250)
19:00:32.310 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - Recover[Token(epoch=0, sequence=3)]: read chain head 1/1
 19:00:32.311 WARN  [test-main] o.c.r.v.StreamsView - append[3]: Overwritten after 3 retries, streams [bee9]
 19:00:42.375 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - fillHole[Token(epoch=0, sequence=4)]: chain head 1/1
 19:00:42.376 INFO  [test-main] o.c.r.v.s.BackpointerStreamView - followBackpointers[bee9@-1]: Found hole at address 4. Starting single step downgrade.
 19:00:42.378 ERROR [LogUnit-Write-Processor-0] o.c.i.BatchWriter - Stream log error. Batch [queue size=0]. StreamLog: [trim mark: 0, tails: TailsResponse(logTail=4, streamTails={})].
 org.corfudb.runtime.exceptions.OverwriteException: null
	at org.corfudb.infrastructure.log.InMemoryStreamLog.throwLogUnitExceptionsIfNecessary(InMemoryStreamLog.java:97)
	at org.corfudb.infrastructure.log.InMemoryStreamLog.append(InMemoryStreamLog.java:60)
	at org.corfudb.infrastructure.BatchWriter.batchWriteProcessor(BatchWriter.java:250)
19:00:42.378 WARN  [test-main] o.c.r.v.r.ChainReplicationProtocol - Recover[Token(epoch=0, sequence=4)]: read chain head 1/1
 19:00:42.379 WARN  [test-main] o.c.r.v.StreamsView - append[4]: Overwritten after 4 retries, streams [bee9]
 19:00:42.379 ERROR [test-main] o.c.r.v.StreamsView - append[5]: failed after 5 retries , streams [bee9], write size 256 bytes
 19:00:42.380 WARN  [test-main] o.c.r.v.ObjectsView - TXEnd[TX[f425]] Aborted Exception {}
 org.corfudb.runtime.exceptions.TransactionAbortedException: TX ABORT  | Snapshot Time = Token(epoch=0, sequence=4) | Failed Transaction ID = 5104398c-e127-45b7-8293-c0cce920f425 | Offending Address = -1 | Conflict Key =  | Conflict Stream = 00000000-0000-0000-0000-000000000000 | Cause = OVERWRITE | Time = 50495 ms | Message = null
	at org.corfudb.runtime.object.transactions.OptimisticTransactionalContext.getConflictSetAndCommit(OptimisticTransactionalContext.java:308)
	at org.corfudb.runtime.object.transactions.OptimisticTransactionalContext.commitTransaction(OptimisticTransactionalContext.java:247)
	at org.corfudb.runtime.view.ObjectsView.TXEnd(ObjectsView.java:161)
Caused by: org.corfudb.runtime.exceptions.AppendException: null
	at org.corfudb.runtime.view.StreamsView.append(StreamsView.java:218)
	at org.corfudb.runtime.view.StreamsView.append(StreamsView.java:228)
	at org.corfudb.runtime.object.transactions.OptimisticTransactionalContext.getConflictSetAndCommit(OptimisticTransactionalContext.java:298)

java.util.concurrent.CompletionException: java.util.concurrent.TimeoutException

	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308)
	at java.util.concurrent.CompletableFuture.orApply(CompletableFuture.java:1371)
	at java.util.concurrent.CompletableFuture$OrApply.tryFire(CompletableFuture.java:1350)
	at java.util.concurrent.CompletableFuture$CoCompletion.tryFire(CompletableFuture.java:1020)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at org.corfudb.util.CFUtils.lambda$failAfter$0(CFUtils.java:109)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.TimeoutException
	at org.corfudb.util.CFUtils.<clinit>(CFUtils.java:97)
	at org.corfudb.AbstractCorfuTest$1$1.evaluate(AbstractCorfuTest.java:133)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)


Process finished with exit code 255

```

--- a/test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionContextTest.java
+++ b/test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionContextTest.java
@@ -4,6 +4,7 @@ import static org.assertj.core.api.Assertions.assertThat;
 import com.google.common.reflect.TypeToken;
 
 import java.nio.ByteBuffer;
+import java.time.Duration;
 import java.util.Map;
 import java.util.UUID;
 import java.util.stream.Collectors;
@@ -14,6 +15,7 @@ import lombok.Data;
 import org.corfudb.protocols.wireprotocol.CorfuMsgType;
 import org.corfudb.protocols.wireprotocol.Token;
 import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuRuntime.CorfuRuntimeParameters;
 import org.corfudb.runtime.clients.TestRule;
 import org.corfudb.runtime.collections.SMRMap;
 import org.corfudb.runtime.exceptions.AbortCause;
@@ -123,9 +125,18 @@ public class OptimisticTransactionContextTest extends AbstractTransactionContext
      */
     @Test
     public void checkSlowWriterTxAbortsOnHoleFill() {
+
+        // The default hole fill timeout duration is 10 seconds. In case of aborting a transaction with cause
+        // overwrite, 5 write attempts are made. This totals to 50 seconds.
+        // Modifying this timeout to 1 second to avoid test timeout after 1 minute.
+        final Duration holeFillTimeout = Duration.ofSeconds(1);
         UUID streamID = UUID.randomUUID();
         CorfuRuntime rtWriter = getDefaultRuntime();
-        CorfuRuntime rtReader = getDefaultRuntime();
+        CorfuRuntime rtReader = getNewRuntime(CorfuRuntimeParameters.builder()
+                .holeFillTimeout(holeFillTimeout)
+                .build())
+                .parseConfigurationString(getDefaultConfigurationString())
+                .connect();
 
         Map<String, String> map = rtWriter
                 .getObjectsView().build()

